### PR TITLE
Add *private* built-in JSX renderer

### PR DESCRIPTION
**Pull request by natemoo-re**
> ## 🚧 This built-in renderer is not be publicly exposed (for the time being).

This renderer is the final step we need to support [MDX](https://mdxjs.com/) pages.

The future implications here are enormous, but the application is intentionally scoped down in this PR. This renderer can only be enabled at the moment by adding an `@astrojs/mdx` integration (which is coming in a follow-up PR).

## Changes

- Adds `astro/jsx-runtime` to core.
- Adds internal JSX renderer to core. This renderer will be automatically enabled if `@astrojs/mdx` is enabled.

## Testing

– Basic test suite added, want to add better coverage over time

## Docs

Intentionally not documenting this, it will be documented as part of MDX support

### CODE REVIEW

**Comment by okikio** [comment]
Exciting

File: packages/astro/src/core/config.ts [hunk 1]
```diff
@@ -346,6 +346,15 @@ export async function validateConfig(
 			adapter: undefined,
 		},
 	};
+	if (
+		// TODO: expose @astrojs/mdx package
```
Comment (matthewp) [inline packages/astro/src/core/config.ts:350]: Is this a TODO now (in this PR) or in the future?

**Review by matthewp — APPROVED** [review]

File: packages/astro/src/core/config.ts (hunk 1 above)
Comment (okikio) [reply packages/astro/src/core/config.ts:350]: @nate is planning to release it today, from what I can gather

Comment (natemoo-re) [reply packages/astro/src/core/config.ts:350]: Great question @matthewp! Coming in a follow-up PR today.

**Pull request merged**